RELEASES.md

```diff
@@ -1,3 +1,109 @@
+Version 1.41.0 (2020-01-30)
+===========================
+
+Language
+--------
+
+- [You can now pass type parameters to foreign items when implementing
+  traits.][65879] E.g. You can now write `impl<T> From<Foo> for Vec<T> {}`.
+- [You can now use any receiver in the `self` position.][64325] E.g. you can
+  now write `fn foo(self: Box<Box<Self>>) {}`. Previously only `Self`, `&Self`,
+  `&mut Self`, `Arc<Self>`, `Rc<Self>`, and `Box<Self>` were allowed.
+- [You can now use any valid identifier in a `format_args` macro.][66847]
+  Previously this was a strict subset.
+- [You can now use `!`/"Never type" in any type position.][65355]
+
+Compiler
+--------
+
+- [Rustc's parser now allows visibility in trait items and enum variants.][66183]
+  These are still rejected when compiled, however they can be seen and parsed
+  by procedural macros.
+- [Rustc will now warn if you have unused `'loop` labels.][66325]
+- [Removed support for the `i686-unknown-dragonfly` target.][67255]
+- [Added tier 3 support\* for the `riscv64gc-unknown-linux-gnu` target.][66661]
+- [You can now pass an arguments file passing the `@path` syntax
+  to rustc.][66172]
+- [You can now provide `--extern` flag without a path, indicating that it is
+  available from the search path or specified with an `-L` flag.][64882]
+
+\* Refer to Rust's [platform support page][forge-platform-support] for more
+information on Rust's tiered platform support.
+
+Libraries
+---------
+
+- [The `core::panic` module is now stable.][66771] This was already stable
+  through `std`.
+- [`NonZero*` numerics now implement `From<NonZero*>` if it's a smaller integer
+  width.][66277] E.g. `NonZeroU16` now implements `From<NonZeroU8>`.
+- [`MaybeUninit<T>` now implements `fmt::Debug`.][65013]
+
+Stabilized APIs
+---------------
+
+- [`Result::map_or`]
+- [`Result::map_or_else`]
+- [`std::rc::Weak::weak_count`]
+- [`std::rc::Weak::strong_count`]
+- [`std::sync::Weak::weak_count`]
+- [`std::sync::Weak::strong_count`]
+
+Cargo
+-----
+
+- [Cargo will now document all the private items for binary crates
+  by default.][cargo/7593]
+- [`cargo-install` will now reinstall the package if it detects that it is out
+  of date.][cargo/7560]
+- [Cargo.lock now uses a more git friendly format that should help to reduce
+  merge conflicts.][cargo/7579]
+- [You can now override specific dependencies's build settings][cargo/7591] E.g.
+  `[profile.dev.overrides.image] opt-level = 2` sets the `image` crate's
+  optimisation level to `2` for debug builds. You can also use
+  `[profile.<profile>.build_overrides]` to override build scripts and
+  their dependencies.
+
+Misc
+----
+
+- [You can now specify `edition` in documentation code blocks to only test code
+  for that edition.][66238] E.g. `edition2018` will only test code when compiled
+  with the 2018 edition.
```

Comment:
Hm, did we really stabilize that? I would expect `edition2018` to *compile* with the 2018 edition, not arbitrarily ignore tests depending on the "global" edition rustdoc is using?

Reply:
PRs linked mention testing. I haven't verified if it's one way or the other.

Reply:
Per https://doc.rust-lang.org/nightly/rustdoc/documentation-tests.html, "edition2018 tells rustdoc that the code sample should be compiled the 2018 edition of Rust. Similarly, you can specify edition2015 to compile the code with the 2015 edition." -- I believe I'm correct, this tests the code in 2018 edition regardless of `--edition` passed to rustdoc itself (i.e., your doc tests can have a different edition from the crate they're in).